tests/cases/fourslash/completionEntryForArrayElementConstrainedToString2.ts

```diff
@@ -0,0 +1,7 @@
+/// <reference path="fourslash.ts" />
+
+//// declare function test<T extends 'a' | 'b'>(a: { foo: T[] }): void
+////
+//// test({ foo: ['a', /*ts*/] })
+
+verify.completions({ marker: ["ts"], includes: ['"a"', '"b"'], isNewIdentifierLocation: true });
```

Comment:
I don't exactly understand what `isNewIdentifierLocation` is about. It's `true` for most of the added test cases but `false` for object property *values*. Is this correct? What's the difference between an object property value and an array element's locations? :thinking:

Reply:
My guess is it's just a bad heuristic that could be patched up?

For reference, it's discussed a bit here: https://github.com/microsoft/TypeScript/issues/42595#issuecomment-771269621

Reply:
I can look into this in a separate PR - it would be cool to have this consistent.